presto-main/src/test/java/com/facebook/presto/operator/aggregation/TestEntropyAggregation.java

```diff
@@ -0,0 +1,177 @@
+    {
+        Long x;
+
+        x = Long.valueOf(1);
+        assertAggregation(entropyFunction,
+                1.0,
+                createLongsBlock(x, null, x));
+
+        x = Long.valueOf(10);
+        assertAggregation(entropyFunction,
+                1.0,
+                createLongsBlock(null, null, x, x));
+    }
+
+    @Test
+    public void entropyOfTwoSkewed()
+    {
+        Long lower = Long.valueOf(30);
+        Long higher = Long.valueOf(70);
+        Double expected = 0.8812908992306931;
+
+        assertAggregation(entropyFunction,
+                expected,
+                createLongsBlock(lower, higher));
+        assertAggregation(entropyFunction,
+                expected,
+                createLongsBlock(higher, lower));
+    }
+
+    @Test
+    public void entropyOfOnlyNulls()
+    {
+        assertAggregation(entropyFunction,
+                0.0,
+                createLongsBlock(null, null));
+    }
+
+    @Test
+    public void entropyOfNegativeCount()
+    {
+        String error = "";
+        try {
+            assertAggregation(entropyFunction,
+                    0.0,
+                    createLongsBlock(Long.valueOf(-1)));
+        }
+        catch (PrestoException e) {
+            error = e.getMessage();
+        }
+        assertTrue(error.contains("negative"));
+    }
+
+    @Override
+    public Block[] getSequenceBlocks(int start, int length)
+    {
+        BlockBuilder blockBuilder = BIGINT.createBlockBuilder(null, length);
+        for (int i = start; i < start + length; i++) {
+            BIGINT.writeLong(blockBuilder, Math.abs(i));
+        }
+        return new Block[] {blockBuilder.build()};
+    }
+
+    @Override
+    public Number getExpectedValue(int start, int length)
+    {
+        final ArrayList<Integer> counts = IntStream
+                .range(start, start + length)
+                .map(c -> Math.abs(c))
+                .boxed()
+                .collect(Collectors.toCollection(ArrayList::new));
+        if (counts.stream().anyMatch(c -> c < 0)) {
+            return null;
+        }
+        final double sum = counts.stream()
+                .mapToDouble(c -> Math.max(c, 0.0))
+                .sum();
+        if (sum == 0) {
+            return 0.0;
+        }
+        final ArrayList<Double> entropies = counts.stream()
+                .filter(c -> c > 0)
+                .map(c -> (c / sum) * Math.log(sum / c))
+                .collect(Collectors.toCollection(ArrayList::new));
+        return entropies.isEmpty() ?
+                0 :
+                entropies.stream().mapToDouble(c -> c).sum() / Math.log(2);
+    }
+
+    @Override
+    protected String getFunctionName()
+    {
+        return TestEntropyAggregation.FUNCTION_NAME;
+    }
+
+    @Override
+    protected List<String> getFunctionParameterTypes()
+    {
+        return ImmutableList.of(StandardTypes.INTEGER);
+    }
+
+    private static final String FUNCTION_NAME = "entropy";
```

Comment:
Move this to the beginning of the class, before `private InternalAggregationFunction entropyFunction;`, and keep a blank line between the two.